typespec-extension/src/code-model-builder.ts

```diff
@@ -1540,114 +1433,157 @@ export class CodeModelBuilder {
     }
   }
 
+  private addParameterOrBodyPropertyToCodeModelRequest(
+    opParameter: SdkPathParameter | SdkHeaderParameter | SdkQueryParameter | SdkBodyModelPropertyType,
+    op: CodeModelOperation,
+    request: Request,
+    schema: ObjectSchema,
+    originalParameter: Parameter,
+  ) {
+    const serializedName = opParameter.serializedName;
+    const existParameter = op.parameters?.find((it) => it.language.default.serializedName === serializedName);
+    request.parameters = request.parameters ?? [];
+    if (existParameter) {
+      // parameter
+      if (
+        existParameter.implementation === ImplementationLocation.Method &&
+        (existParameter.origin?.startsWith("modelerfour:synthesized/") ?? true) &&
+        !(existParameter.schema instanceof ConstantSchema)
+      ) {
+        request.parameters.push(cloneOperationParameter(existParameter));
+      }
+    } else {
+      // property from anonymous model
+      const existBodyProperty = schema.properties?.find((it) => it.serializedName === serializedName);
+      if (existBodyProperty && !existBodyProperty.readOnly && !(existBodyProperty.schema instanceof ConstantSchema)) {
+        request.parameters.push(
+          new VirtualParameter(
+            existBodyProperty.language.default.name,
+            existBodyProperty.language.default.description,
+            existBodyProperty.schema,
+            {
+              originalParameter: originalParameter,
+              targetProperty: existBodyProperty,
+              language: {
+                default: {
+                  serializedName: existBodyProperty.serializedName,
+                },
+              },
+              summary: existBodyProperty.summary,
+              implementation: ImplementationLocation.Method,
+              required: existBodyProperty.required,
+              nullable: existBodyProperty.nullable,
+            },
+          ),
+        );
+      }
+    }
+  }
+
   private findResponseBody(bodyType: Type): Type {
     // find a type that possibly without http metadata like @statusCode
     return this.getEffectiveSchemaType(bodyType);
   }
 
-  private processResponse(op: CodeModelOperation, resp: HttpOperationResponse, longRunning: boolean) {
+  private processResponse(
+    op: CodeModelOperation,
+    statusCode: number | HttpStatusCodeRange | "*",
```

Comment:
Do we have existing test case for `HttpStatusCodeRange` (probably local)?

Reply:
Seems not, I added one in our local, see tsp and generated code in this commit b7ff02c70f5ab694cf858c9304725d1e6bce56ea 
will consider added to cadl-ranch as well.